File: 209_MinimumSizeSubarraySum/minimumSizeSubarraySum.js
var minSubArrayLen = function(target, nums) {
    let sum = 0
    let minWindow = Infinity
    let leftPtr = 0

    for (let rightPtr = 0; rightPtr < nums.length; rightPtr++) {
        sum += nums[rightPtr]
        while (sum >= target) {
            var curWindow = rightPtr - leftPtr + 1
            minWindow = Math.min(minWindow, curWindow)
            sum -= nums[leftPtr]    
            leftPtr += 1
        }
    }
    return minWindow == Infinity ? 0 : minWindow
};

console.log(minSubArrayLen(7, [2, 3, 1, 2, 4, 3]))
console.log(minSubArrayLen(4, [1, 4, 4]))
console.log(minSubArrayLen(11, [1,1,1,1,1,1,1,1]))

File: 209_MinimumSizeSubarraySum/minimumSizeSubarraySum.py
def minSubArrayLen(self, target: int, nums: List[int]) -> int:
    sum = 0
    min_window = float('inf')
    left_ptr = 0

    for right_ptr in range(len(nums)):
        sum += nums[right_ptr]
        while sum >= target:
            cur_window = right_ptr - left_ptr + 1
            min_window = min(min_window, cur_window)
            sum -= nums[left_ptr]    
            left_ptr += 1
    
    return 0 if min_window == float('inf') else min_window

print(minSubArrayLen(7, [2, 3, 1, 2, 4, 3]))
print(minSubArrayLen(4, [1, 4, 4]))
print(minSubArrayLen(11, [1,1,1,1,1,1,1,1]))